src/tests/e2etest/NetworkingTests.cs

```diff
@@ -0,0 +1,104 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using NUnit.Framework;
+using Microsoft.Azure.Devices;
+using System.Text.Json;
+using System;
+using System.Threading.Tasks;
+using System.Text.RegularExpressions;
+using System.Collections.Generic;
+
+namespace e2etesting
+{
+    public class NetworkingTests : E2eTest
+    {
+        string ComponentName = "Networking";
+        string PropertyName = "NetworkConfiguration";
+        public partial class Networking
+        {
+            public string InterfaceTypes { get; set; }
+            public string MacAddresses { get; set; }
+            public string IpAddresses { get; set; }
+            public string SubnetMasks { get; set; }
+            public string DefaultGateways { get; set; }
+            public string DnsServers { get; set; }
+            public string DhcpEnabled { get; set; }
+            public string Enabled { get; set; }
+            public string Connected { get; set; }
+        }
+
+        public void RegexTest(string testString, string patternString)
+        {
+            string splitPattern = ";";
+            Regex regexPattern = new Regex(@patternString);
+            string[] result = Regex.Split(testString, splitPattern);
+            foreach (string piece in result)
+            {
+                Assert.True(regexPattern.IsMatch(piece));
+            }
+        }
+
+        public void RegexTestCommaSplit(string testString, string piecePatternString, string smallPatternString)
+        {
+            string splitPattern = ";";
```

Comment:
Consider changing 'splitPattern' and 'splitPatternComma' to something like 'interfaceDelimiter' and 'propertyDelimiter', respectively.